boards/common/blxxxpill/Makefile.include

```diff
@@ -1,36 +1,9 @@
 INCLUDES += -I$(RIOTBOARD)/common/blxxxpill/include
 
-# define the default port depending on the host OS
-PORT_LINUX ?= /dev/ttyUSB0
-PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.SLAB_USBtoUART*)))
+# For older versions of the DFU bootloader, set DFU_USB_ID to 1d50:6017 and
+# uncomment the DFU_FLAGS line
+DFU_USB_ID ?= 1eaf:0003
+#DFU_FLAGS ?= -s 0x08002000:leave
 
-# setup serial terminal
-include $(RIOTMAKE)/tools/serial.inc.mk
-
-# optionally, use dfu-util to flash via usb
-# note: needs a bootloader flashed before, config below is compatible
-# with blackmagic_dfu, see https://github.com/blacksphere/blackmagic/
-# To stop bootloader from loading an existing firmware, pull down
-# (ground) GPIO B1.
-ifeq ($(PROGRAMMER),dfu-util)
-  export ROM_OFFSET ?= 0x2000 # Skip the space needed by the embedded bootloader
-  FLASHER = dfu-util
-  DEBUGGER = # no debugger
-  RESET ?= # dfu-util has no support for resetting the device
-
-  FLASHFILE ?= $(BINFILE)
-  FFLAGS = -d 1eaf:0003 -a 2 -D $(FLASHFILE)
-  # for older bootloader versions use this:
-  # FFLAGS = -d 1d50:6017 -s 0x08002000:leave -D $(FLASHFILE)
-else
-
-  # this board uses openocd by default
-  DEBUG_ADAPTER ?= stlink
-  STLINK_VERSION ?= 2
```

Comment:
This specific configuration is lost with the current change: the default is to use st-link2.1. I know the distinction no longer exists is OpenOCD master and some people are trying to merge the config (#11824) but the default version of OpenOCD is still 0.10 on release/LTS Linux distros.
At least, this PR should not change this.

Reply:
The bluepill/blackpill has no integrated st-link. Instead, the user will have to provide an external st-link. Thus, to me it makes sense to have it default to version 2.1, which is the more common hardware. Users that use an old versions of the st-link and *at the same time* an old version OpenOCD can still set this manually.

Reply:
I totally with how we should deal with ST-Link, don't worry. It's just that the default version is changed. What I'm saying is that, for clarity, this should be in its own commit or in its own PR. Because for the moment, the commit doesn't mention the reason why it's removed (the default st-link version).

Reply:
I added the change of the ST-Link version as own commit on top and added the reasoning to the commit message details.

Reply:
> Thus, to me it makes sense to have it default to version 2.1, which is the more common hardware.

Is it though? The common cheap [ST-Link](https://www.ebay.de/sch/i.html?_nkw=st-link) probe reports as  `STLINK V2J34S7 (API v2) VID:PID 0483:3748`.

Unfortunately it seems like both by ST-Link (v2) probes are broken (can't flash with either config) but replacement from China is on it's way.

I wish OpenOCD would make another release already so we could just use `interface/stlink.cfg` and not care about the version difference.

Reply:
Hmm, I have the same cheap programmers and they are (or at least were prioir to the BMP upgrade) v2.1. I bought like 40 of those for students from different vendors and every vendor had his own PCB design with a different pinout. It was quite a mess to make Black Magic Probes out of them due to the different pinouts. Apparently they also differ in software...

I'll adapt to keep it v2 for now and we can later clean up, if OpenOCD ever starts making new releases again.

Reply:
If it works with those probes with `STLINK_VERSION ?= 2-1` than you can keep it - I'm just saying I can't flash it with either version with my probes, so I can't tell if it makes a difference.

Reply:
Sounds like the reset signal is not properly send, likely due a misconfiguration in the ST-Link firmware. I had the same issue with mine. Try pressing and holding the reset button and release the button when OpenOCD is trying to connect the bluepill during "make flash". That should work